### DISCUSSION TITLE
Is BlackHole compatible with Apple Silicon?

### DISCUSSION THREAD

**Discussion by QianNangong** [ExistentialAudio/BlackHole#227] [Feature Requests]

**Comment by devinroth** [comment]
Not yet. Do you have a machine? Are you willing to test it out?

**Reply by bryanvrgsc** [reply]
I'm willing to test it out.

**Comment by QianNangong** [comment]
I'll get the new MacBook Pro in few days, and I'm willing to test it out.
I'm not sure whether Big Sur on ARM has more restrictions than Intel's.

**Comment by devinroth** [comment]
Awesome. Well let me know when you get it and we’ll try it out.

Devin

**Comment by devinroth** [comment]
Try this out. Or if anyone else has the new M1 chip please test. 

[BlackHole.driver.zip](https://github.com/ExistentialAudio/BlackHole/files/5558242/BlackHole.driver.zip)

**Comment by QianNangong** [comment]
My new Mac will ship next week...

**Comment by devinroth** [comment]
Great. Turns out it was really easy to build it for the M1. Let me know how it goes when you get the new Mac.

**Comment by illegalmp3** [comment]
Hey there @devinroth ,

I just came across BlackHole while searching for an iShowU Audio Capture replacement, and I appreciate you making a build for the new M1 Macs. I received my M1 MBP yesterday.

I installed the driver from the zip you provided, and unfortunately, I experienced the distorted/slurred audio that someone else reported elsewhere in this Issues section. I noticed that the new build you provided was the 16ch version of Blackhole, so I uninstalled it and installed the 2ch (non-M1) version. 2ch works perfectly with OBS on my M1 MBP, sending audio from a music app to a Multi-Output Device. I'll definitely be snagging a 2ch M1 build, should you get around to it soon. :)

**Comment by devinroth** [comment]
Wow. So the 2ch non M1 version works. I'm quit impressed. As I'm sure you saw in the other thread, the issues with the 16ch are unfortunately not a BlackHole issue. I'll post a 2ch version here for you to test out.

**Comment by devinroth** [comment]
[BlackHole2ch.driver.zip](https://github.com/ExistentialAudio/BlackHole/files/5562458/BlackHole2ch.driver.zip)

**Comment by illegalmp3** [comment]
Huh! Well, I downloaded the BlackHole2ch.driver.zip version you provided just above this post, overwrote the previous non-M1 2ch driver, restarted CoreAudio using the terminal language under the 'Uninstall' instructions, and now I'm experiencing the same slurring/distortion that occurred with the 16ch M1 version.

I'm gonna try removing the driver altogether and 'clean' installing the 2ch M1 driver, restart CoreAudio after that and see if it works properly.

**Comment by devinroth** [comment]
Is it in fact the 2ch one? Just double check that I didn’t accidentally give you the wrong one.

Devin

**Comment by devinroth** [comment]
May have miss labeled it.

Devin

**Comment by illegalmp3** [comment]
Just double checked, and you indeed gave me a 2ch version of BlackHole in BlackHole2ch.driver.zip.

My 'clean install' idea didn't work, since manually pasting the contents of the above .zip into Library/Audio/Plug-Ins/HAL and restarting CoreAudio via Terminal kept giving me one of these security messages.

<img width="260" alt="Screen Shot 2020-11-18 at 1 49 15 PM" src="https://user-images.githubusercontent.com/38093668/99592690-62458d00-29a5-11eb-8b69-6bb381651746.png">

I've seen this message plenty of times before, went into the Security settings to tell macOS to 'Open Anyway', but doing so just opened a new Finder window for BlackHole2ch.driver in the HAL folder. Because of that, the above security message just kept displaying itself every time I restarted CoreAudio.

So I removed BlackHole using the Uninstaller pkg, then used BlackHole2ch.v0.2.7.pkg to install the non-M1 2ch version, tested that with a Multi-Output Device in OBS, still worked great. I manually pasted the contents from your BlackHole2ch.driver.zip over the existing non-M1 driver, restarted CoreAudio, and with the M1 2ch driver, the slurring/distortion has returned when sending audio through a Multi-Output Device.

**Comment by devinroth** [comment]
I probably forgot to sign the driver. I'll try again soon.

**Comment by devinroth** [comment]
Try this one.
[BlackHole2ch.driver.zip](https://github.com/ExistentialAudio/BlackHole/files/5563675/BlackHole2ch.driver.zip)

**Comment by illegalmp3** [comment]
Unfortunately, I'm experiencing the same behavior as before with the new .zip you provided. I tried all the same things; overwriting the previous non-M1 driver with the new driver produced the same slurring and distortion, and installing the new driver fresh with the previous one removed still gave me that Security pop-up, even after I tried rebooting immediately after dropping the new driver into the HAL folder.

**Comment by devinroth** [comment]
Could you send me a clip of the slurring/distortion you are experiencing?

**Comment by devinroth** [comment]
What version of the driver are you on?

Devin

**Comment by devinroth** [comment]
Try the latest M1 build from this thread and report back.

Devin

**Comment by CestDiego** [comment]
After trying last build from thread I can confirm that this does not work for me either :'(  When I select the multiple output to listen to I hear noises and sounds garbled up.

**Comment by devinroth** [comment]
@CestDiego perhaps it's the multi-output that is the problem. Could you run a test with only BlackHole?

**Comment by illegalmp3** [comment]
@devinroth My apologies, I just got back to my laptop after some time away and I’ll be taking some clips of the results I’m getting with the M1 2ch build after some family time this evening.

Really, I just wanted to jump in and say that it was _incredibly_ obvious to me that @devinroth was joking about accepting donations of a Mac Mini, but I guess even winky smileys at the end of a sentence aren’t clear enough for _all_ people. 😂

**Comment by devinroth** [comment]
Thanks for the support @illegalmp3. I'm just going to delete all of those comments as they are off topic.

**Comment by devinroth** [comment]
Could someone with an M1 try building BlackHole on their machine and report back?

**Comment by CestDiego** [comment]
I tried using only blackhole 2ch as audio out (without multi out like before, although I need multi out to record screencasts with internal sound+Microphone). If I play YouTube videos, it speeds them up and I cannot hear anything. Weird

**Comment by devinroth** [comment]
I've been working on some bug fixes. I wonder if they will solve some of the M1 issues. 

Anyone willing to try building the driver from the 0.2.8 branch and see how it goes?

**Comment by wookayin** [comment]
@devinroth I tried building the 0.2.8 branch, placed the driver file into HAL, but Blackhole does not show up on M1. I am not an expert app developer; do you have any idea how can we troubleshoot or look at the log files?

**Comment by devinroth** [comment]
Did you run `sudo launchctl kickstart -kp system/com.apple.audio.coreaudiod` in terminal?

**Comment by wookayin** [comment]
Yes I did. It was my bad -- I was supposed to copy the driver, not sym-linking (it had worked for me before). After that, Blackhole shows up. However it seems it is not working: I cannot hear any sound (maybe it is outputting some non-audible sounds) and the volume bar fluctuates randomly.

**Comment by devinroth** [comment]
BlackHole doesn’t output sounds. Create a multi output and try that.

Devin

**Comment by wookayin** [comment]
When I say "I cannot hear any sound" I meant one after multi output settings or a sound recording from the Blackhole's virtual microphone input. Actually, the only thing I hear is some crackles.

Actually I somehow posted a wrong screenshot, sorry. Here is the weird "fluctuation" behavior:

![blackhole-test-1127-2](https://user-images.githubusercontent.com/1009873/100493972-dcdf7c80-310a-11eb-8a27-113d73db045d.gif)

**Comment by devinroth** [comment]
Hmmm.  Not sure what to make of that.

**Comment by z1r0-** [comment]
I'm usually using multiple Blackhole Drivers with SoundSource (to redirect app output to different channels) and WaveLink (to mix volume of the channels with streamdeck). It works with x64 version of the blackhole driver (i compiled on my old mac). If I recompile blackhole driver on m1 macbook the sound is broken.

To eliminate possible problems with wavelink (there is no arm version yet) and other software closed them all. set my output to the Blackhole driver and used Music app to play some music and quicktime to record the audio-output of the blackhole device. The result is the same noisy output. 

> BlackHole doesn’t output sounds. Create a multi output and try that.

I didn't manage to create a multi-output or whatever in this midi-stuff which I'm able to output over my speaker. Just don't know how to do it. But the test described above was the closest I could get to "with only Blackhole".

btw. I used 0.2.8 branch to build the driver.

**Comment by devinroth** [comment]
Thanks for the update.

I think I will need to test it on an M1 myself. I’ll take a Month or so to get here.

Devin

**Comment by dougluslee** [comment]
I have downloaded the installer package and installed 2 channel BH on a new M1 MacBook Air and can report the download installed and has been running without issue. Recording browser streams with Audacity.  Blackhole 2ch appears in the mac sound drop down, audio preferences, midi app, all related menus/boxes, etc. Its all there. Audacity running fine on ‘rosetta’ and smoking various tasks. The Soundflower ‘kext’ failed to make the jump to the new pc. A fresh install of soundflower is not supported in Big Sur so I am delighted to find this replacement. Thanks very Much!

**Reply by wookayin** [reply]
Interesting. On my M1 mac, Blackhole does not work for me either on rosetta or arm64. Probably because I have built Blackhole myself.

**Reply by wookayin** [reply]
Can confirm the download version (0.2.7 2ch) works well on M1 under Rosetta. Only self-compiled one doesn't.

**Comment by calebcbaker** [comment]
> Thanks for the update. I think I will need to test it on an M1 myself. I’ll take a Month or so to get here. Devin

@devinroth Does this require an Apple Developer account? If not I can attempt to compile on mine.

**Comment by devinroth** [comment]
No you don’t need a dev account.

Devin

**Comment by devinroth** [comment]
Just an update on this. BlackHole works fine with Rosetta but there are bugs when compiled for Apple Silicon. I spent some time figuring out those bugs and will release the next version of BlackHole to run on Apple Silicon natively.

**Reply by z1r0-** [reply]
Tried to build the current 0.2.8 branch on Apple Silicon and can confirm that it works now 👍 great job

**Comment by stevebanik** [comment]
There are still some weird issues going on with the latest release on my M1 mini.  I can't tell if it's Quicktime or something else.  

When I playback Spotify, it works for a while then "silently" crashes as I'm recording with Quicktime.  After it fails in that manner, I try to open a YouTube video in Chrome and it won't playback.  If I reboot, it's fixed for awhile, then the problem reappears.  

If I use Music.app to playback the song and record with Quicktime, it works for awhile then starts looping a short 1s (or so) segment over and over.

The only solution for me at this point is to reboot.  When I'm recording Quicktime, I can't tell if the problem is occurring until I stop recording and listen to the screen recording.

Any idea what's going on?

**Comment by devinroth** [comment]
The problem from what I’ve noticed is Multi-Outputs. They will work for 20 minutes than then crash. Are you using a multi-output?

Devin

**Comment by stevebanik** [comment]
Yes.

**Comment by devinroth** [comment]
We’ll have to find an alternative.

This should work.

System Output to BlackHole
Audacity input to BlackHole output to speaker
Make sure you enable monitoring in Audacity.
And then input to QT to BlackHole.

Let me know if that works for you.

Devin

**Comment by devinroth** [comment]
I just ran a test with a multi-output. If the built-in output is the first on the list it doesn't seen to have any issues.

**Comment by devinroth** [comment]
Don't ask me why.

**Comment by stevebanik** [comment]
I got it to work on the M1 mini as follows:

Create multi-output device in midi setup.  Select mac mini speakers, lg
monitor speakers, behringer umc404 and blackhole 2ch, check the box for
drift correction.  Master device is UMC.  Right click and set as output
(but oddly this was not possible and there is an exclamation mark there.)


In control panel, set Multi-Output device as output.  Set Blackhole 2ch as
input.


In Ableton, select Blackhole 2ch as input.  Select Multi-Output device as
output.


In Quicktime, select Blackhole 2ch, start recording.

**Comment by stevebanik** [comment]
One edit: in Ableton I had to select UMC404 as the input, not Blackhole
2ch.

I also noticed that Blackhole 16ch only records mono in QT as others have
noted, so when I switched to 2ch I got stereo.

**Comment by z1r0-** [comment]
Hm.. I had this "stops working after 20 mins" bug too with 0.2.8 branch build but thought it was not blackhole related because the virtual audio driver by reincubate (apple silicon, comes with camo) had the same problem. I use multi-device output but blackhole isn't a part of it (use it for different bluetooth/non bluetooth speakers so I don't have to switch the output when changing the device I'm using). I'm using blackhole as channels for wave link software (which isn't updated to apple silicon yet) and thought the problem is related to this piece of software since rosetta version of blackhole worked without issues.
After 0.2.8 was merged I recompiled my blackhole drivers from master branch and out of the sudden all problems were gone. There was no update for wave link or any other software I'm using so I am as confused as you why it suddenly stopped crashing, but happy that it works stable now. Thought maybe there were additional changes to blackhole after my 0.2.8 build but don't know.

**Comment by devinroth** [comment]
No additional changes. What did you do different?

Someone said the problem goes away when drift correction is off.

Devin

**Comment by z1r0-** [comment]
I just retested the reincubate audio driver and the issue seems to be gone there too. So guess it's not related to the rebuild. Maybe it is drift correction. I dunno remember if I had it on before and turned it off at that moment. But I played around with it as far as I remember and currently it's not active for any device.

**Comment by DWilson-Audio** [comment]
1. I am running latest Version 0.2.8  2 channel version

2. I have one instance of blackhole with 2 in and 2 out.  No multi-output or aggregate.

3. Roon Labs audio writes too Blackhole IN.  This appears to work just fine for hours on end with no use of the Blackhole OUT.

4. When I enable Blackhole OUT to a VST Host such as kushview Element or Apple GarageBand but NOT send the audio out to any device the system will hang after typically 20 minutes but it varies in time a little bit. In both these applications I have no plugs in running.  If I enable the VST output to a USB Bryston DAC it hangs as well.    There are no error messages presented and the CPU / Memory resources are all well below any meaningful threshold. 

6. If I send Roon out directly to the same USB DAC bypassing Blackhole and a VST host the system works fine. 

If any one has some suggestions to try I would be happy to try them.

**Comment by devinroth** [comment]
I think I figured it out. I'll post a beta version here.

**Comment by devinroth** [comment]
Alright everyone. Please try this out. I'm pretty sure it solves the 20 minute crash issue.

[BlackHole2ch.v0.2.9-beta.pkg.zip](https://github.com/ExistentialAudio/BlackHole/files/5871937/BlackHole2ch.v0.2.9-beta.pkg.zip)

**Reply by bryanvrgsc** [reply]
This is a beta version?

**Reply by devinroth** [reply]
Yeah. If it fixes the issue I'll release it. If not, back to the drawing board.

**Reply by DWilson-Audio** [reply]
Hi Devin,  

Your new version works flawlessly.  I have been running it for over 3 hours at 192k without any hiccups.  Many thanks for the very fast turnaround on the problem.  Much appreciated.  I will be posting the Blackhole solution for Dirac room correction to the Roon Labs community in a day or so. 

Cheers
Drew

**Reply by bryanvrgsc** [reply]
I have been running it for over more than 24 hours, there's no problem. Nice job. 👍

**Reply by DWilson-Audio** [reply]
48 hours continuous use.  No issues under several M1 configurations.  Many thanks as well.

**Comment by rcarrascoz** [comment]
Hello devinroth, I am Rafa, the same that I wrote you 5 minutes ago from gmail. I am going to check it right now, almos install it. I will say you how it is going... Sorry for my English :)

**Comment by rcarrascoz** [comment]
devinroth, I have a question, why do I need to put in apple audio settings out multiple (blackhole and headset) to get work on obs studio? It is not confortable after when I want loud down or up my headset... Moreover, I can set in obs settings on audio -> desktop sound -> only a blackhole2h...

thanks.

**Reply by devinroth** [reply]
I'm not super familiar with OBS. Maybe someone else can chime. You'll want a multi-output as your system output and BlackHole as the input into OBS.

**Comment by rcarrascoz** [comment]
Ok but obs is the best tool to stream to youtube or twitch, ahd blackhole
is awesome for that.
But, ok, now it is working fine... well yesterday I lost desktop sound, but
after that I can restore again. I recorded a game during 1h and all video
has sound :)

**Comment by devinroth** [comment]
Great. I’ve pushed the changes to the master branch and released it.

Devin

**Comment by Clifford-Yen** [comment]
Am I the only one who still encounter the buzzing noise even after updating to v0.2.9? Although it does not happen every time, the audio of recordings still distorts at some points (20 minutes, 30minutes, you name it) with the combination of the built-in multi-output device and QuickTime screen recording. Anyone?

The only reliable way to record and monitor the sound at the same time is still using the monitoring function of Audacity.

**Comment by devinroth** [comment]
Are you using drift correction? If not, you probably should.

Devin

**Reply by Clifford-Yen** [reply]
Hi @devinroth, so I should enable drift correction on BlackHole 2ch? Let me try it.

**Reply by Clifford-Yen** [reply]
OK so I can confirm enabling drift correction seems to fix the problem. I will update if I find something wrong. Thank you @devinroth !

**Reply by calebcbaker** [reply]
I'm experiencing the same thing on v0.2.9 with drift correction on BlackHole 2ch, it seems to take awhile to appear, sometimes it's 20min other times I can go an entire day. This morning I was listening to spotifiy and started hearing static, paused spotify, static went away, resumed and it came back. The static starts off very slowly, just a few pops here and there but as time goes on it gets exponentially worse. 

You only hear it on playback though, I did a recording with Audacity and it didn't pick up the static at all. To temporarily get rid of the static all I have to do is change the output to something else and switch it back to the multi-output and it sounds fine again.

**Comment by devinroth** [comment]
Test out the beta version (attached to thread) . See if it’s different.

Devin

**Reply by calebcbaker** [reply]
That's the one I've been using, should I try the non-beta version?

**Reply by calebcbaker** [reply]
Sorry for the late reply, after testing I'm still getting static on both versions. To be sure I ran the uninstaller and rebooted for each test.

**Reply by devinroth** [reply]
And you have drift compensation on?

**Comment by devinroth** [comment]
Yeah.

Devin

**Comment by Sleightly2015** [comment]
Hello devinroth. I've joined GitHub specifically to join in this discussion to try to help as a user. Here is what I'm working with:

MacBook Pro 13", M1, Big Sur
BlackHole v0.2.9

I want to use BlackHole to route audio to/from applications, including Reaper DAW, Steam, OBS, and Zoom.
After following the wiki pages to create a Multi-Output Device and an Aggregate Device, I have not been able to route any audio to/from anywhere. Additionally, YouTube often stops working while I'm trying to troubleshoot. As soon as I return my computer's sound input/output back to MacBook Pro, everything works fine again.
Let me know how I can be helpful. Perhaps even to Zoom and I can share my screen? I'm super unfamiliar with GitHub, so you'll have to guide me a little here...

**Comment by devinroth** [comment]
First thing, try using the 2ch version. See if that makes a difference and report back.

Devin

**Reply by Sleightly2015** [reply]
I rebooted my computer and am using an Aggregate Device that includes BlackHole 2ch. YouTube is playing fine again, but I'm unable to route audio to either Reaper DAW nor Zoom.
<img width="653" alt="Screenshot 2021-02-02 at 19 57 56" src="https://user-images.githubusercontent.com/78440738/106648739-f4b30100-6590-11eb-872e-0663d06149da.png">

**Comment by devinroth** [comment]
Put on drift correction.

Use BlackHole as the input to Zoom / DAW.

Devin

**Reply by Sleightly2015** [reply]
No dice. Screenshots were taken while playing YouTube video. You can see no response in microphone level in Zoom settings and no waves being recorded in DAW.

<img width="799" alt="Zoom Screenshot 2021-02-02 at 20 02 59" src="https://user-images.githubusercontent.com/78440738/106649570-106ad700-6592-11eb-91cd-61c6d8c96412.png">
<img width="1437" alt="Reaper Recording Screenshot 2021-02-02 at 20 04 04" src="https://user-images.githubusercontent.com/78440738/106649572-12349a80-6592-11eb-9bf7-00e62e0a9494.png">
<img width="719" alt="Reaper Audio Settings Screenshot 2021-02-02 at 20 04 20" src="https://user-images.githubusercontent.com/78440738/106649577-13fe5e00-6592-11eb-846d-638bdb19838c.png">

**Reply by Sleightly2015** [reply]
Standby. There is life in Zoom after one setting change in Sound menu: Selecting BlackHole 2ch as Output instead of Aggregate Device. Reaper wasn't helped at all...

<img width="1440" alt="YouTube:Zoom Screenshot 2021-02-02 at 20 10 07" src="https://user-images.githubusercontent.com/78440738/106650334-144b2900-6593-11eb-9788-fbe0391d84b8.png">
<img width="668" alt="Sound Settings Screenshot 2021-02-02 at 20 10 49" src="https://user-images.githubusercontent.com/78440738/106650353-19a87380-6593-11eb-9215-b809245403f3.png">
<img width="763" alt="Audio MIDI Setup Screenshot 2021-02-02 at 20 11 00" src="https://user-images.githubusercontent.com/78440738/106650357-1a410a00-6593-11eb-9c29-f3548de750a5.png">

**Comment by devinroth** [comment]
Just do a simple test.

System Output to BlackHole2ch
Reaper Input BlackHole2ch

Devin

**Comment by devinroth** [comment]
Outputing to an aggregate doesn’t output to all the devices in the list. Just to the first one.

You need to use a multi-output instead.

Devin

**Reply by Sleightly2015** [reply]
Made the change. Zoom still successful. Reaper still getting nothing.

<img width="763" alt="Audio MIDI Setup Screenshot 2021-02-02 at 20 20 58" src="https://user-images.githubusercontent.com/78440738/106652019-30e86080-6595-11eb-9b90-615551c154a8.png">
<img width="780" alt="Sound Settings Screenshot 2021-02-02 at 20 21 05" src="https://user-images.githubusercontent.com/78440738/106652033-347be780-6595-11eb-863e-0f395d86a3cc.png">
<img width="1440" alt="YouTube Zoom Screenshot 2021-02-02 at 20 23 43" src="https://user-images.githubusercontent.com/78440738/106652041-3645ab00-6595-11eb-9f09-93f4b025a563.png">
<img width="1437" alt="Reaper Screenshot 2021-02-02 at 20 25 09" src="https://user-images.githubusercontent.com/78440738/106652057-39d93200-6595-11eb-9738-b865b5d20969.png">

**Comment by devinroth** [comment]
Does reaper have mic access?

Devin

**Reply by Sleightly2015** [reply]
I believe this is the way to give microphone access. I now separated input and output sources, but that did not help. For reference, here is how I learned to record computer audio in Reaper (instructions are specifically for SoundFlower, but I expect settings to be the same).

On a positive note, I also tested OBS and audio looks successful there!

<img width="1505" alt="OBS Screenshot 2021-02-02 at 20 32 16" src="https://user-images.githubusercontent.com/78440738/106653004-8b35f100-6596-11eb-9419-b26c7a167c28.png">
<img width="1437" alt="Reaper Screenshot 2021-02-02 at 20 34 59" src="https://user-images.githubusercontent.com/78440738/106653017-8ec97800-6596-11eb-8d62-5d108724e295.png">

**Comment by devinroth** [comment]
Can you record from your mic?

Devin

**Comment by devinroth** [comment]
No.

Check System Preferences -> Security & Privacy -> Privacy -> Microphone. If iReaper is not listed it won’t be able to record.

If it’s not listed, you need to contact Reaper and get it figured out.

Devin

**Reply by Sleightly2015** [reply]
You're right, it's not there.

**Reply by Sleightly2015** [reply]
Thanks for the input, @devinroth. I'll follow up with Reaper and see what they come up with. If it's updated and able to work, I'll update here. I'll also keep testing it around other applications.

**Reply by Sleightly2015** [reply]
Update: It is a bug within Reaper's version for macOS ARM64 (beta)

To workaround:
1. Install Intel (10.15+) version, run program and it will immediately ask for microphone permission. Grant permission.
2. Install ARM version. Delete Intel version. 
3. Confirm that microphone permission has transferred to ARM version (You'll know because the name in System Preferences > Security & Privacy > Privacy > Microphone will have changed from "Reaper64" (Intel version) to "Reaper" (ARM version). This happened for me with no additional step necessary.

Using the same settings as before that worked with Zoom and OBS (see pictures and comments above), and I can confirm that audio is now being routed to within Reaper.

-- Celebratory screenshot 🥳 --

<img width="1548" alt="Reaper Screenshot 2021-02-03 at 10 03 52" src="https://user-images.githubusercontent.com/78440738/106723297-3f6f6000-6607-11eb-92f9-b9dbca856dae.png">

**Reply by logiota** [reply]
Nevermind, new release is here!
Can I support using crypto?
Thank you all!

**Reply by devinroth** [reply]
Sure can. I use BTC ETH or LTC. Let me know which one and I'll give you an address.

**Comment by NathanWolf** [comment]
Hello, sorry if this unnecessarily pings everyone here, I realize this thread hasn't been touched in a month ... I read through this entire discussion, but I'm still a little unclear on the state of the M1 build.

Did it make its way into an official release with an installer? Does it seem to be working and stable?

I am willing to do some testing, but this would be my first experience with BlackHole (coming from Soundflower) so I'm unfamiliar with all of the installation details and issues that came up in this thread and may not be much help as a tester.

Thank you for your time!

**Comment by devinroth** [comment]
Yes the latest release is stable for M1.

Devin

**Reply by NathanWolf** [reply]
Thank you! And I apologize, now that I've run the installer and see that it says so right in the first dialog.

I might suggest a small note on the download page about how there is native M1 support, but of course up to you. Thanks again :)

**Comment by vicosfe** [comment]
Hey! I use a macbook pro m1, everything works fine, but on my touchbar, when I select a multi-output device, the volume control of the output sound disappears

**Comment by devinroth** [comment]
That’s because multi outputs don’t have volume.

Devin

**Comment by andysocial21** [comment]
Hi
Am using Blackhole 2ch on M1 Mac Mini to dictate audio from VLC movie to Textedit - works fine for about 5 lines and then times out - am I doing something wrong? (but very promising so far - thanks)

**Comment by dforsee** [comment]
I'm using M1 Macs (Pro and Max) on Monterey  Looks like it partially works but not sending the audio out any channel except 1 (on the 16 ch version).  Could be my application ShowKontrol but doesn't appear to have that issue with any other sound outputs.  Feel free to count me in on betas...

**Comment by devinroth** [comment]
Does ShowKontrol work with 16ch properly? How about the 2ch?

It doesn’t make any sense that only 1 channel works.

**Comment by Guitarbarian** [comment]
Hey, love your work! I've upgraded to OBS 29 as it handles the M1 chip a lot better. I've unfortunately got a new issue with audio. Blackhole sets up and works perfectly for around 15 minutes before crackling out and obs stops picking it up, if I reselect my multi output device from my DAW it comes back to life. I've tried changing around some settings in OBS but haven't found a solution yet. It could be an OBS 29 issue but I'm not smart enough to work it out.

**Comment by devinroth** [comment]
Read the faq. Probably audio drift.

Devin

**Comment by stevebanik** [comment]
I wonder if the "crackling" is related to the [well-known issue](https://www.reddit.com/r/macbookpro/comments/tlqpc0/my_fix_for_the_2021_m1_macbook_speaker/) on M1 MacBook Pros.  There are a variety of "solutions" but some of them don't work.  To my knowledge, Apple has still not acknowledged the issue, but I could be wrong on that as I haven't looked recently.
